modules/cloud-run/variables.tf

```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "audit_log_triggers" {
+  description = "Event arc triggers (Audit log)"
+  type = list(object({
+    service_name = string
+    method_name  = string
+  }))
+  default = null
+}
+
+variable "containers" {
+  description = "Containers"
+  type = list(object({
+    image   = string
+    command = list(string)
+    args    = list(string)
+    env     = map(string)
+    env_from = map(object({
+      key  = string
+      name = string
+    }))
+    resources = object({
+      limits = object({
+        cpu    = string
+        memory = string
+      })
+      requests = object({
+        cpu    = string
+        memory = string
+      })
+    })
+    ports = list(object({
+      name           = string
+      protocol       = string
+      container_port = string
+    }))
+    volume_mounts = map(string)
+  }))
+}
+
+variable "iam" {
+  description = "IAM bindings for Cloud Run service in {ROLE => [MEMBERS]} format."
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "ingress_settings" {
+  description = "Ingress settings"
+  type        = string
+  default     = null
+}
+
+variable "labels" {
+  description = "Resource labels"
+  type        = map(string)
+  default     = {}
+}
+
+variable "name" {
+  description = "Name used for cloud run service"
+  type        = string
+}
+
+variable "prefix" {
+  description = "Optional prefix used for resource names."
+  type        = string
+  default     = null
+}
+
+variable "project_id" {
+  description = "Project id used for all resources."
+  type        = string
+}
+
+variable "pubsub_triggers" {
+  description = "Eventarc triggers (Pub/Sub)"
+  type        = list(string)
+  default     = null
+}
+
+variable "region" {
+  description = "Region used for all resources."
+  type        = string
+  default     = "europe-west1"
+}
+
+variable "revision_name" {
+  description = "Revision name"
+  type        = string
+  default     = null
+}
+
+variable "service_account" {
+  description = "Service account email. Unused if service account is auto-created."
+  type        = string
+  default     = null
+}
+
+variable "service_account_create" {
+  description = "Auto-create service account."
+  type        = bool
+  default     = false
+}
+
+variable "traffic" {
+  description = "Traffic"
+  type        = map(number)
+  default     = null
+}
+
+variable "volumes" {
+  description = "Volumes"
+  type = list(object({
+    name        = string
+    secret_name = string
+    items = list(object({
+      key  = string
+      path = string
+    }))
+  }))
+  default = null
+}
+
+variable "vpc_connector_config" {
+  description = "VPC connector configuration. Set `create_config` attributes to trigger creation."
+  type = object({
+    egress_settings = string
+    name            = string
+    ip_cidr_range   = string
```

Comment:
I remember discussions with you the fact the `ip_cidr_range` and `network` might be dynamic, passed in from other modules' outputs, and the plan to split this into two variables to avoid issues.

Can we split this into two variables, somthing like a boolean `vpc_connector_create` which you then use in `count` on the resource, and `vpc_connector_config` which has the actual variables? I would also skip `name` and just use the top-level name.

Reply:
split done